st2reactor/st2reactor/container/sensor_wrapper.py

```diff
@@ -67,6 +75,120 @@ def dispatch(self, trigger, payload=None):
         """
         self._dispatcher.dispatch(trigger, payload=payload)
 
+    def get_value(self, name):
+        """
+        Retrieve a value from the datastore for the provided key.
+
+        :param name: Key name.
+        :type name: ``str``
+
+        :rtype: ``str`` or ``None``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        self._logger.audit('Retrieving value from the datastore (name=%s)', name)
+
+        try:
+            kvp = client.keys.get_by_id(id=name)
+        except Exception:
+            return None
+
+        if kvp:
+            return kvp.value
+
+        return None
+
+    def set_value(self, name, value, ttl=None):
```

Comment:
Can you take the KVPApi object here instead? Basically as we add more features to this object the parameters to set_value will increase and could get quite unwieldy?